### DISCUSSION TITLE
Does this not work with the free version of the Twitter API?

### DISCUSSION THREAD

**Discussion by geolectric** [tweepy/tweepy#2119] [Questions] [answered]
I just tried to post a Tweet but get the following error message:

```
tweepy.errors.Forbidden: 403 Forbidden
453 - You currently have access to Twitter API v2 endpoints and limited v1.1 endpoints only. If you need access to this endpoint, you may need a different access level. You can learn more here: https://developer.twitter.com/en/docs/twitter-api/getting-started/about-twitter-api#v2-access-level
```

Is it not possible to use this library with the free version of the Twitter API?

**Comment by Harmon758** [comment]
Tweepy should be fully functional with Free access.
You'll need to provide the relevant code and/or full traceback if you need further help.

**Comment by geolectric** [comment, marked as answer]
@Harmon758 Yeah, I ended up getting it, there's just a lot of old information out there.

This was the most helpful page I eventually found in getting it to work:
https://docs.tweepy.org/en/stable/examples.html#examples

If you're using the free version of the Twitter API, then go to the `API v2 tab`, and follow the `Authentication` instructions.

And then still under the `API v2 tab`, but follow the `Create Tweet` instructions, this was the part that got me:

```
# The app and the corresponding credentials must have the Write permission

# Check the App permissions section of the Settings tab of your app, under the
# Twitter Developer Portal Projects & Apps page at
# https://developer.twitter.com/en/portal/projects-and-apps

# Make sure to reauthorize your app / regenerate your access token and secret 
# after setting the Write permission
```

I had already created my Access Token, and it only had Read permissions, I had to `Revoke` it and then `Regenerate` it for the Read/Write permissions to finally show up, but it's all working now.

**Reply by Harmon758** [reply]
Tweepy's documentation itself should be generally up to date in regards to the latest versions.
Recent API changes aren't necessarily reflected yet simply because there's still a lot of flux and unknowns due to lack of communication on Twitter's end.

**Comment by ihabpalamino** [comment]
hello i was trying to get data of tweet using username and dates since until but i got this error message;forbidden-453-you-currently-have-access-to-twitter-api-v2-endpoints-and-limited-v1-1-endpoints

**Reply by Harmon758** [reply]
As that error indicates, that's not available with Free access, and that's not relevant to this discussion.
